convert2rhel/unit_tests/cert_test.py

```diff
@@ -16,77 +16,108 @@
-        self.assertEqual(len(cert.loggerinst.critical_msgs), 1)
-        # Remove the temporary directory tree
+
+    monkeypatch.setattr(utils, "DATA_DIR", data_dir)
+    monkeypatch.setattr(system_info, "arch", request.param["arch"])
+
+    yield rhel_cert_dir
+
+    # Remove the temporary directory tree if we created it earlier
+    if request.param["data_dir"] is not None:
         shutil.rmtree(os.path.join(utils.DATA_DIR, "rhel-certs"))
 
-    @unit_tests.mock(cert, "loggerinst", unit_tests.GetLoggerMocked())
-    @unit_tests.mock(utils, "DATA_DIR", unit_tests.NONEXISTING_DIR)
-    @unit_tests.mock(system_info, "arch", "nonexisting_arch")
-    def test_get_cert_path_nonexisting_dir(self):
-        self.assertRaises(SystemExit, cert.SystemCert._get_cert)
-        self.assertEqual(len(cert.loggerinst.critical_msgs), 1)
-
-    @unit_tests.mock(tool_opts, "arch", "x86_64")
-    @unit_tests.mock(utils, "DATA_DIR", os.path.join(base_data_dir, "6", "x86_64"))
-    def test_install_cert(self):
-        # By initializing the cert object we get a path to an existing
-        # certificate based on the mocked parameters above
-        system_cert = cert.SystemCert()
-        system_cert._target_cert_dir = unit_tests.TMP_DIR
-
-        system_cert.install()
-
-        installed_cert_path = os.path.join(system_cert._target_cert_dir, system_cert._cert_filename)
-        self.assertTrue(os.path.exists(installed_cert_path))
-        shutil.rmtree(unit_tests.TMP_DIR)
+
+@pytest.mark.parametrize(
+    (
+        "message",
+        "cert_dir",
+    ),
+    (
+        pytest.param(
+            "Error: System certificate (.pem) not found in %(cert_dir)s.",
+            {
+                "data_dir": unit_tests.TMP_DIR,
+                "arch": "arch",
+            },
+            id="missing-certificate",
+        ),
+        pytest.param(
+            "Error: Could not access %(cert_dir)s.",
+            {
+                "data_dir": None,
+                "arch": "nonexisting_arch",
+            },
+            id="missing-cert-dir",
+        ),
+    ),
+    indirect=("cert_dir",),
+)
+def test_get_cert_path_missing_cert(message, caplog, cert_dir):
+    # Check response for the non-existing certificate in the temporary dir
+    with pytest.raises(SystemExit):
+        cert.SystemCert._get_cert()
+    assert len(caplog.records) == 1
+    assert caplog.records[0].levelname == "CRITICAL"
+    assert caplog.records[0].message == message % {"cert_dir": cert_dir}
+
+
+@pytest.mark.parametrize(
+    ("arch", "rhel_version", "pem"),
+    (
+        ("x86_64", "6", "69.pem"),
+        ("x86_64", "7", "69.pem"),
+        ("x86_64", "8", "479.pem"),
+        ("ppc64", "7", "74.pem"),
+    ),
+)
+def test_get_cert_path(arch, rhel_version, pem, monkeypatch):
+    monkeypatch.setattr(utils, "DATA_DIR", unit_tests.TMP_DIR)
+    # Check there are certificates for all the supported RHEL variants
+    utils.DATA_DIR = os.path.join(BASE_DATA_DIR, rhel_version, arch)
```

Comment:
We could just mock the `DATA_DIR` here with the line `100`, I think... Probably we don't need to mock to a temporary dir and then overwrite that with the value we want, right?

Reply:
@abadger, pushed the fix here.